### DISCUSSION TITLE
Custom loss function. how to get the number of predicted boxes of 1 image in "fast_rcnn.py" ?

### DISCUSSION THREAD

**Discussion by bmohammed0227** [facebookresearch/detectron2#2988] [Q&A]
Hello, i'm trying to add a custom loss function to train the model to avoid intersection between the boxes and to get exactly 17 boxes, so i followed the solution proposed in Issue#974. (there is another solution in the documentation but i don't want to write a new subclass like mentionned, i just want to add new informations to the default loss function).

The problem is that the number of boxes (i mean the value of the variable "proposal_boxes") is always the same so i cannot converge on 17 boxes, apparently it's linked with the configuration (ROI_HEADS.BATCH_SIZE_PER_IMAGE), i didn't use this configuration but i still get 512 boxes, am i using the wrong variable to get the number of predicted boxes on 1 image ?

also i'm trying to apply the new modifications on this line, am i correct ?
`losses_ = {
            "loss_cls": my_new_loss_Function,`

Thanks !
